mongoengine/document.py

```diff
@@ -488,7 +488,13 @@ def reload(self):
         """
         id_field = self._meta['id_field']
         collection = self._get_collection()
-        son = collection.find_one(self._db_object_key)
+        # If this is a lazy object, we only have the ID field and don't want to
+        # call _db_object_key, since _db_object_key could fetch (reload) the
+        # object.
+        if self._lazy:
+            son = collection.find_one({ '_id': self.pk })
+        else:
+            son = collection.find_one(self._db_object_key)
```

Comment:
Actually there is no great way to do this right now (there might be when we merge with the official MongoEngine repo) since we don't know for sure if the value wasn't loaded, or if it's just `None`. For now, you just have to be really careful with `only()` (which you already know). As I already mentioned though, this isn't a new issue.